### PR TITLE
New provisioning changes

### PR DESCRIPTION
**Pull request by ananth-at-camphor-networks**

### CODE REVIEW

File: cluster/aws/config-default.sh [hunk 1]
```diff
@@ -82,6 +82,8 @@ ENABLE_CLUSTER_DNS="${KUBE_ENABLE_CLUSTER_DNS:-true}"
 DNS_SERVER_IP="10.0.0.10"
 DNS_DOMAIN="cluster.local"
 DNS_REPLICAS=1
+OPENCONTRAIL_PUBLIC_SUBNET="${OPENCONTRAIL_PUBLIC_SUBNET:-10.1.0.0/16}"
+OVERLAY_NETWORK_PROVIDER="${OVERLAY_NETWORK_PROVIDER:-none}" # opencontrail
```
Comment (pedro-r-marques) [inline cluster/aws/config-default.sh]: This should probably be "NETWORK_PROVIDER". Whether it is overlay or not... there will be non-overlay options (e.g. VLANs, calico).

File: cluster/saltbase/salt/kube-network-manager/init.sls [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/etc/kubernetes/manifests/kube-network-manager.manifest:
+  file.managed:
+    - source: https://raw.githubusercontent.com/rombie/contrail-kubernetes/manifests/cluster/kube-network-manager.manifest
```
Comment (pedro-r-marques) [inline cluster/saltbase/salt/kube-network-manager/init.sls]: Lets merge this into the Juniper repository... with an extension manifest.sls

File: cluster/saltbase/salt/top.sls [hunk 3]
```diff
@@ -17,7 +17,11 @@ base:
     - cadvisor
     - kube-client-tools
     - kubelet
+{% if pillar.get('overlay_network_provider', '').lower() == 'opencontrail' %}
```
Comment (pedro-r-marques) [inline cluster/saltbase/salt/top.sls]: network_provider

**Comment by pedro-r-marques** [comment]
The commit is targeting master... don't merge into master.
We need a branch that we can then use to attempt to upstream into the kubernetes repository.
Lets not merge anything into master.

**Comment by ananth-at-cloud-spectra-dot-ai** [comment]
Yes. I raise done to have a easy way to look at the diffs. master will not
be touched.

Regards,
Ananth

**Pull request closed without merging**